assets/js/modules/ads/datastore/settings.test.js

```diff
@@ -73,6 +80,38 @@ describe( 'modules/ads settings', () => {
 				},
 			} );
 		} );
+
+		it( 'should send a POST request to FPM settings endpoint when toggle state is changed', async () => {
+			const fpmEndpoint = new RegExp(
+				'^/google-site-kit/v1/core/site/data/fpm-settings'
+			);
+
+			fetchMock.postOnce( settingsEndpoint, ( url, opts ) => ( {
+				body: JSON.parse( opts.body )?.data,
+				status: 200,
+			} ) );
+
+			fetchMock.postOnce( fpmEndpoint, {
+				body: JSON.stringify( {
+					data: { settings: { isEnabled: true } },
+				} ),
+				status: 200,
+			} );
+
+			await registry
+				.dispatch( CORE_SITE )
+				.setFirstPartyModeEnabled( true );
```

Comment:
We don't need to `await` to set this local state.
```suggestion
			registry.dispatch( CORE_SITE ).setFirstPartyModeEnabled( true );
```